lib/inherit.js

```diff
@@ -84,7 +79,14 @@ function override(base, res, add) {
 
                         return res;
                     };
-                result.__base = baseMethod;
+
+                result.__base = function() {
+                    var baseMethod = base[name]
```

Comment:
Admittedly, I don't like an idea to have such checks in runtime. Probably we need to introduce dev/production modes.